sdk/resourcemanager/azure-resourcemanager-network/src/main/java/com/azure/resourcemanager/network/implementation/WebApplicationFirewallPolicyImpl.java

```diff
@@ -0,0 +1,241 @@
+    public boolean isEnabled() {
+        return WebApplicationFirewallEnabledState.ENABLED.equals(innerModel().policySettings().state());
+    }
+
+    @Override
+    public List<ApplicationGateway> getAssociatedApplicationGateways() {
+        return getAssociatedApplicationGatewaysAsync().collectList().block();
+    }
+
+    @Override
+    public Flux<ApplicationGateway> getAssociatedApplicationGatewaysAsync() {
+        if (CoreUtils.isNullOrEmpty(this.innerModel().applicationGateways())) {
+            return Flux.empty();
+        }
+        return Flux.fromIterable(this.innerModel().applicationGateways())
+            .flatMap(applicationGatewayInner -> manager().applicationGateways().getByIdAsync(applicationGatewayInner.id()));
+    }
+
+    @Override
+    public WebApplicationFirewallPolicyImpl withDetectionMode() {
+        return withMode(WebApplicationFirewallMode.DETECTION);
+    }
+
+    @Override
+    public WebApplicationFirewallPolicyImpl withPreventionMode() {
+        return withMode(WebApplicationFirewallMode.PREVENTION);
+    }
+
+    @Override
+    public WebApplicationFirewallPolicyImpl withMode(WebApplicationFirewallMode mode) {
+        ensurePolicySettings().withMode(mode);
+        return this;
+    }
+
+    @Override
+    public WebApplicationFirewallPolicyImpl enablePolicy() {
+        ensurePolicySettings().withState(WebApplicationFirewallEnabledState.ENABLED);
+        return this;
+    }
+
+    @Override
+    public WebApplicationFirewallPolicyImpl disablePolicy() {
+        ensurePolicySettings().withState(WebApplicationFirewallEnabledState.DISABLED);
+        return this;
+    }
+
+    @Override
+    public WebApplicationFirewallPolicyImpl withBotProtection() {
+        boolean existBotDetection = ensureManagedRules()
+            .managedRuleSets()
+            .stream()
+            .anyMatch(ruleSet -> BOT_DETECTION_RULE_SET_TYPE.equals(ruleSet.ruleSetType()));
+        if (!existBotDetection) {
+            ensureManagedRules().managedRuleSets().add(
+                new ManagedRuleSet()
+                    .withRuleSetType(BOT_DETECTION_RULE_SET_TYPE)
+                    .withRuleSetVersion(BOT_DETECTION_RULE_SET_VERSION_DEFAULT));
+        }
+        return this;
+    }
+
+    @Override
+    public WebApplicationFirewallPolicyImpl withoutBotProtection() {
+        ensureManagedRules().managedRuleSets()
+            .removeIf(ruleSet -> BOT_DETECTION_RULE_SET_TYPE.equals(ruleSet.ruleSetType()));
+        return this;
+    }
+
+    @Override
+    public WebApplicationFirewallPolicyImpl withBotProtection(String version) {
+        if (version == null) {
+            return withBotProtection();
+        } else {
+            ensureManagedRules()
+                .managedRuleSets()
+                .stream()
+                .filter(ruleSet -> BOT_DETECTION_RULE_SET_TYPE.equals(ruleSet.ruleSetType()))
+                .findFirst()
+                .ifPresent(ruleSet -> ruleSet.withRuleSetVersion(version));
```

Comment:
what if not found? should we add the rule?

Reply:
Thanks. `withBotProtection()` should call `withBotProtection(version)` in the implementation, not the other way around.